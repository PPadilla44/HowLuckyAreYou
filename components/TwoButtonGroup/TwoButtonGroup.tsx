import { StyleSheet } from 'react-native'
import React, { Dispatch, FC, SetStateAction } from 'react'
import { ButtonGroup } from '../Themed'

interface Props {
    buttons: any[];
    selectedIndex: number;
    setSelectedIndex: Dispatch<SetStateAction<number>>;
}

const TwoButtonGroup: FC<Props> = ({ buttons, selectedIndex, setSelectedIndex }) => {



    return (
        <ButtonGroup
            buttons={buttons}
            onPress={(newIndex: number) => setSelectedIndex(newIndex)}
            selectedIndex={selectedIndex}
            containerStyle={styles.container}
            buttonStyle={styles.button} 
            textStyle={styles.text}
            selectedButtonStyle={styles.selectedButton}
            selectedTextStyle={styles.selectedText}
            innerBorderStyle={{color: "transparent"}}
            />
    )
}

export default TwoButtonGroup

const styles = StyleSheet.create({
    container: {
        padding: 2,
        height: 32,
        borderRadius: 9,
        marginBottom: 20,
        marginTop: 0
    },
    button: {
        borderRadius: 7,
        opacity: .5
    },
    text: {
        fontFamily: "Futura",
        fontSize: 15,
        fontWeight: "bold",
        color: "black"
    },
    selectedButton: {
        backgroundColor: "white",
        opacity: 1
    },
    selectedText: {
        color: "black"
    }
})