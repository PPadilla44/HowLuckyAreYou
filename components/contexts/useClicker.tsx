import React, { createContext, Dispatch, ReactNode, useContext, useReducer } from 'react';
import { Action, initialClickerState, ClickerState } from '../../store/clicker';
import clickerReducer from "../../store"

const ClickerContext = createContext<{ state: ClickerState, dispatch?: Dispatch<Action>  }>({ state: initialClickerState });

export const useClicker = () => {
    return useContext(ClickerContext);
}

type Props = {
    children: ReactNode;
};

export const ClickerProvider = ({ children }: Props) => {

    const [state, dispatch] = useReducer(clickerReducer, initialClickerState);

    const providerState = {
        state,
        dispatch
    }

    return (
        <ClickerContext.Provider value={providerState}>
            {children}
        </ClickerContext.Provider>
    )
}
