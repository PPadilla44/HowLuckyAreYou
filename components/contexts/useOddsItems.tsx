import React, { createContext, Dispatch, ReactNode, useContext, useReducer } from 'react';
import { Action,  } from '../../store/oddsItems';
import oddsItemsReducer from "../../store/oddsItems"
import { OddsItemInterface } from '../../types';

const OddsItems = createContext<{ state: OddsItemInterface[], dispatch?: Dispatch<Action>  }>({ state: [] });

export const useOddsItems = () => {
    return useContext(OddsItems);
}

type Props = {
    children: ReactNode;
};

export const OddsItemsProvider = ({ children }: Props) => {

    const [state, dispatch] = useReducer(oddsItemsReducer, [] );

    const providerState = {
        state,
        dispatch
    }

    return (
        <OddsItems.Provider value={providerState}>
            {children}
        </OddsItems.Provider>
    )
}
