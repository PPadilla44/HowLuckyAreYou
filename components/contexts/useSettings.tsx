import { createContext, Dispatch, ReactNode, useContext, useReducer } from "react";
import { Appearance } from "react-native";
import settingsReducer from "../../store/settings"
import { Action, initialSettingsState, SettingsState } from "../../store/settings";

const Settings = createContext<{ state: SettingsState , dispatch?: Dispatch<Action> }>( { state: initialSettingsState } );

export const useSettings = () => {
    return useContext(Settings);
}


type Props = {
    children: ReactNode;
};

export const SettingsProvider = ({ children }: Props) => {

    const [state, dispatch] = useReducer(settingsReducer, initialSettingsState );

    if(state.data.appearance.fromDevice) {
        state.data.appearance.appearance = Appearance.getColorScheme() as "light" | "dark";
    }

    const providerState = {
        state,
        dispatch
    }

    return (
        <Settings.Provider value={providerState}>
            {children}
        </Settings.Provider>
    )
}
