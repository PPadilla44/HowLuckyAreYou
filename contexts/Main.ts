import { useReducer } from "react";
import { View } from "react-native";
import CustomContext from "./context";

export interface MainState {
    count: number;
    odds: number;
    didHit: boolean;
    results: {
        BtnColor: string;
        text: string;
    };
}

export enum MainActionKind {
    INCREASE = "INCREASE",
}

export interface Action {
    type: MainActionKind,
    payload?: any
}

export const initialMainState = {
    count: 0,
    odds: .1,
    didHit: false,
    results: {
        BtnColor: "",
        text: ""
    }
}

function increment(state: MainState) {
    console.log("****************");

    const tempCount = state.count + 1;

    const max = 1 / (state.odds);

    const randomNumFromMax = Math.ceil(Math.random() * max);

    console.log(randomNumFromMax, max);

    if (randomNumFromMax === max) {
        console.log("GOT IT");
        const userOdds = tempCount / max;
        let tempResults
        if (userOdds < 1) {
            tempResults = { BtnColor: "green", text: "YOU ARE REALLY LUCKY" };
        } else if (userOdds > 1) {
            tempResults = { BtnColor: "red", text: "YOU ARE NOT LUCKY" };
        } else {
            tempResults = { BtnColor: "blue", text: "YOU ARE normal" };
        }
        const newState = {
            ...state,
            count: tempCount,
            results: tempResults,
            didHit: true
        }
        console.log(newState);

        return newState
    }

    return {
        ...state,
        count: tempCount
    }
}

export function mainReducer(state: MainState = initialMainState, action: Action) {
    const { type, payload } = action;
    switch (type) {
        case MainActionKind.INCREASE:
            return increment(state)

        default:
            return state;
    }
}

// const ParentComponent = ({ children }) => {

//     const [userState, usersDispatch] = useReducer(mainReducer, initialMainState);

//     const providerState = {
//         userState,
//         usersDispatch
//     }

//     return (
//         <View>
//         </View>
//     )
// }

// export default ParentComponent;