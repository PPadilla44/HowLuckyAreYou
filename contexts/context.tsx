import React, { createContext, useContext } from 'react';

const CustomContext = createContext(null);

export function useCustomContext() {
    return useContext(CustomContext);
}

export default CustomContext;