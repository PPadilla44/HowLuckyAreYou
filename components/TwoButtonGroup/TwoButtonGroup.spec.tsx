import { cleanup, fireEvent, RenderAPI, render } from "@testing-library/react-native";
import React from "react";
import { ClickerProvider } from "../contexts/useClicker";
import TwoButtonGroup from "./TwoButtonGroup";

describe('<TwoButtonGroup />', () => {
    const setStateMock = jest.fn();

    const props = {
        buttons: ["First", "Second"],
    }

    afterEach(cleanup)

    let wrapper: RenderAPI;
    beforeEach(() => {
        wrapper = render(
            <ClickerProvider>
                <TwoButtonGroup
                    {...props}
                />
            </ClickerProvider>
        )
    });

    it("should render correctly", () => {
        wrapper.getByText("First");
        wrapper.getByText("Second");
        const buttons = wrapper.getAllByTestId("buttonGroupItem");
        expect(buttons).toHaveLength(2);
        expect(buttons[0].children[0]).toHaveStyle({
            opacity: 1
        })
        expect(buttons[1].children[0]).toHaveStyle({
            opacity: .5
        })
        expect(buttons[0]).toHaveTextContent("First")
        expect(buttons[1]).toHaveTextContent("Second")
    });

    it('should be able to switch between buttons', () => {
        const [button1, button2] = wrapper.getAllByTestId("buttonGroupItem");
        fireEvent.press(button2);
        fireEvent.press(button1);
    });


})