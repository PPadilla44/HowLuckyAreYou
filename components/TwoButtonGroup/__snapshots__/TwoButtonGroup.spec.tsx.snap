// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TwoButtonGroup /> should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#808080",
      "borderColor": "#808080",
      "borderRadius": 9,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 32,
      "marginBottom": 20,
      "marginHorizontal": 10,
      "marginTop": 0,
      "marginVertical": 5,
      "overflow": "hidden",
      "padding": 2,
    }
  }
>
  <View
    style={
      Object {
        "borderRightColor": "transparent",
        "borderRightWidth": 1,
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
      testID="buttonGroupItem"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 7,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontFamily": "Futura",
              "fontSize": 15,
              "fontWeight": "bold",
            }
          }
          testID="buttonGroupItemText"
        >
          First
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
      testID="buttonGroupItem"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 7,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 0.5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontFamily": "Futura",
              "fontSize": 15,
              "fontWeight": "bold",
            }
          }
          testID="buttonGroupItemText"
        >
          Second
        </Text>
      </View>
    </View>
  </View>
</View>
`;
